### DISCUSSION TITLE
[Solved] Local models don't acknowledge context + current file

### DISCUSSION THREAD

**Discussion by cxar** [yetone/avante.nvim#850] [General]
Seems like local models don't receive the context + current file if the file is too long, or just fail to acknowledge it. Seems to work for smaller files or remote models. For some reason it also doesn't work for highlights. I believe the long context may be causing these smaller local models to get confused. 

It could be nice to have a config to limit the max code context size sent to the model.

Opening this discussion to see if others have experienced this + have any workarounds

**Update: I had to set the ollama max context to match the model's max context using a [Modelfile](https://github.com/ollama/ollama/blob/main/docs/modelfile.md). It is now working as-expected.**

**Comment by MaxOSchulte** [comment]
Sounds similar to: https://github.com/yetone/avante.nvim/discussions/592

**Comment by rheinardkorf** [comment]
@cxar , any chance you can give an example?

**Reply by cxar** [reply]
Sure!

The problem was due to Ollama default modelfiles using the [default context window of 2048 tokens](https://github.com/ollama/ollama/blob/main/docs/modelfile.md#:~:text=Sets%20the%20size%20of%20the%20context%20window%20used%20to%20generate%20the%20next%20token.%20(Default%3A%202048)). I have some larger code files (on the order of a few hundred LoC), and when conversing using Avante it claimed it could not read the file. 

This is solvable by either setting the `num_ctx` field in the request to match the model's max context ([although there's some debate about whether that is an actual solution](https://github.com/ollama/ollama/issues/2714)), or by using a custom Modelfile. 

Here's the content of an example modelfile:
```
FROM qwen2.5-coder:32b-instruct-q8_0
PARAMETER num_ctx 32768
```

You then load that into Ollama like so ([docs](https://github.com/ollama/ollama/blob/main/docs/modelfile.md#basic-modelfile)):
`ollama create choose-a-model-name -f <location of the file e.g. ./Modelfile>`

and use the new model name in Avante.

**Reply by rheinardkorf** [reply]
Thanks. That solved my issue and I have my qwen2.5 14b model working well.
